crates/runtime/src/user_agent_util.rs

```diff
@@ -0,0 +1,40 @@
+use regex::Regex;
+use std::sync::LazyLock;
+
+static UA_REGEX: LazyLock<Result<Regex, regex::Error>> =
+    LazyLock::new(|| Regex::new(r"^(\S+) (\S+) \((.*?)\)$"));
+
+pub fn extract_user_agent(ua_header: Option<&str>) -> (String, String, String) {
+    let ua_header = ua_header.unwrap_or("");
+    if let Ok(regex) = &*UA_REGEX {
+        if let Some(ua_parts) = regex.captures(ua_header) {
```

Comment:
Does this need to be a regex? We control all of the possible inputs here, would a simple string split work? Regexes can add non-trivial perf cost, and this is in the critical path of requests.

Reply:
someone could pass a wrong header if they use `curl` or something else. I think a regex makes sense, because we can match for format and expected values.

Reply:
Sure, but someone using curl isn't what we care about. We want to know our SDK usage.